clj-refactor.el

```diff
@@ -1947,6 +1947,43 @@ FEATURE is either :clj or :cljs."
                                                (list "clj" "cljs"))
                         "clj"))))))
 
+(defun cljr--beginning-of-reader-conditional ()
+  "Return position of beginning of containing reader conditional.
+
+Otherwise `nil' if outside of reader conditional."
+  (save-excursion
+    (let ((reader-start nil))
+      (while (not (or reader-start (cljr--top-level-p)))
+        (backward-up-list)
```

Comment:
I would prefer if we used more the builtin functions instead of depending on paredit. In this case it depends on `cljr--depth-at-point`, which depends on `paredit-current-parse-state` (which is a soft wrapper around `beginning-of-defun` and `parse-partial-sexp` both of which are builtin) and `paredit-in-string-p`, which also depends on parse state from `parse-partial-sexp`. As discussed below, it appears that `cljr--top-level-p` doesn't handle files with preceeding unmatched parentheses, if we need to address this, I think we would need to dig into the `syntax-ppss` functions emacs provides in `syntax.el`.

If there are specific edge cases that paredit-backwards-up handles over backward-up-list then that would make sense to stay with paredit, but I would prefer to use the out of the box sexp navigation if it suffices. I think it makes a lot of sense to use paredit when manipulating sexps as it adds a lot to ensure the file remains parseable, I'm just not aware of the advantage for simple sexp navigation.